applications/tari_dan_app_utilities/src/base_layer_scanner.rs

```diff
@@ -356,23 +361,39 @@ impl BaseLayerScanner {
         Ok(())
     }
 
-    fn register_burnt_utxo(&mut self, output: &TransactionOutput) -> Result<(), BaseLayerScannerError> {
+    async fn register_burnt_utxo(&mut self, output: &TransactionOutput) -> Result<(), BaseLayerScannerError> {
         let address = SubstateAddress::UnclaimedConfidentialOutput(
             UnclaimedConfidentialOutputAddress::try_from_commitment(output.commitment.as_bytes()).map_err(|e|
                 // Technically impossible, but anyway
                 BaseLayerScannerError::InvalidSideChainUtxoResponse(format!("Invalid commitment: {}", e)))?,
         );
 
-        let substate = Substate::new(
-            0,
-            SubstateValue::UnclaimedConfidentialOutput(UnclaimedConfidentialOutput {
-                commitment: output.commitment.clone(),
-                encrypted_data: EncryptedData(output.encrypted_data.to_bytes()),
-            }),
-        );
-        let shard_id = ShardId::from_address(&address, 0);
-        self.shard_store
-            .with_write_tx(|tx| tx.save_burnt_utxo(&substate, address, shard_id))
+        let substate = SubstateValue::UnclaimedConfidentialOutput(UnclaimedConfidentialOutput {
+            commitment: output.commitment.clone(),
+            encrypted_data: EncryptedData(output.encrypted_data.to_bytes()),
+        });
+        let epoch = self.epoch_manager.current_epoch().await?;
+        self.state_store
```

Comment:
I wonder if these should be proposed by the leader of the local committee as well?

Reply:
yeah I think that makes sense, at some point we need to fix this. If only the shard(UnclaimedOutpt) stores it, it could be enough however on the claim transaction, every validator involved would need to validate that they have seen the output on the base layer. Maybe we can have "validator transactions" or "system transactions" so that validators can propose things like a new unclaimed output, but not sure if that is necessary in this case